fe/fe-core/src/main/java/org/apache/doris/nereids/parser/trino/TrinoFnCallTransformers.java

```diff
@@ -0,0 +1,84 @@
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.parser.trino;
+
+import org.apache.doris.nereids.analyzer.UnboundFunction;
+import org.apache.doris.nereids.parser.ParserContext;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.functions.Function;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The builder and factory for {@link org.apache.doris.nereids.parser.trino.TrinoFnCallTransformer},
+ * and supply transform facade ability.
+ */
+public class TrinoFnCallTransformers {
+
+    private static final ImmutableListMultimap<String, TrinoFnCallTransformer> TRANSFORMER_MAP;
+    private static final ImmutableListMultimap.Builder<String, TrinoFnCallTransformer> transformerBuilder =
+            ImmutableListMultimap.builder();
+
+    static {
+        registerTransformers();
+        TRANSFORMER_MAP = transformerBuilder.build();
+    }
+
+    private TrinoFnCallTransformers() {
+    }
+
+    /**
+     * Function transform facade
+     */
+    public static Function transform(String sourceFnName, List<Expression> targetFnArgs, ParserContext context) {
+        List<TrinoFnCallTransformer> transformers = getTransformers(sourceFnName);
+        for (TrinoFnCallTransformer transformer : transformers) {
+            if (transformer.check(targetFnArgs, context)) {
+                return transformer.transform(targetFnArgs, context);
+            }
+        }
+        return null;
+    }
+
+    private static List<TrinoFnCallTransformer> getTransformers(String sourceFnName) {
+        ImmutableList<TrinoFnCallTransformer> trinoFunctionCallTransformers =
+                TRANSFORMER_MAP.get(sourceFnName);
+        return trinoFunctionCallTransformers == null ? ImmutableList.of() : trinoFunctionCallTransformers;
+    }
+
+    private static void registerTransformers() {
+        registerStringFunctionTransformer();
+        // TODO: add other type function transformer
+    }
+
+    private static void registerStringFunctionTransformer() {
+        doRegister("codepoint", 1, "ascii");
+    }
+
+    private static void doRegister(
```

Comment:
hwo to register varArg function?

Reply:
This is a good question, we should add source method(trino or other) args desc when register. I think this maybe require a function sign args mapping not only the args num. WDYT?